### DISCUSSION TITLE
Failed to compile Jamulus

### DISCUSSION THREAD

**Discussion by SamueleStefanoni** [jamulussoftware/jamulus#1432] [Technical Support & Help]
Hey everybody!
I'm a super fan and happy user of Jamulus and also a Qt developer who aims to contribute to this beautiful project.

Unfortunately, I tried all day long to setup my environment to compile and debug Jamulus with no luck.
I followed the instructions and I ended up with this setup:
- Windows 10 machine 64bit
- Qt 6.0.3 with msvc2019_64 + QtCreator 4.14
- Microsoft Visual Studio Build Tools 2019 with MSVC 1.42 x64 + Windows 10 SDK + CMake + Debug functionalities

After a long journey now Qt is autodetecting both MSVC compiler and debugger, so everything should work fine.
I try to compile Jamulus and I receive this message:

```
C:\cygwin64\DevPI\workspace_qt\jamulus\src\recorder\cwavestream.h(35): error C2027: using undeclared type 'QString'
..\..\..\..\Qt\6.0.3\msvc2019_64\include\QtCore/qsysinfo.h(52): note: check 'QString' declaration
```
.... and other 22 same entries around other source files...

Of course, I tried also to add QString include, but with no luck. I tried to compile a new project with QString class and of course, is working fine. Is there an explanation for that behavior?

I tried also to compile with MinGW 5.12.2 (with which I usually worked in Qt env) and things are going slightly better: I'm able to compile, but not to debug since the first main() line breakpoint is enabled but the execution is not stopping. Function stack shows this message: "Binary debug information has not been found for this frame" (of course I compiled in debug build option and this setup has been working fine for several projects. 
Hope that someone will give me some good advice and let me enjoy this code.
Cheers!

**Comment by hoffie** [comment]
Hi,

Qt 6 is not (yet?) supported. Can you try with the latest 5.x release?
Also, did you follow some kind of documentation which may need clarification regarding this?

Kind regards,
Christian

**Reply by SamueleStefanoni** [reply]
Hey Christian!
yes you are right! In fact using Qt 5.15 got me to a successful compile! Great!
Now, I think I need a last hint as, once compiled in debug mode, I receive this message: " This does not seems a debug build". And, in fact, I can't debug because breakpoint are not stopping the code. What do you think am I doing wrong?
 
![image](https://user-images.githubusercontent.com/77231821/113447206-afc51280-93f9-11eb-99b0-d32c1a39b571.png)

**Reply by hoffie** [reply]
Glad to hear. Sadly, I don't have any experience with Qt Creator.

Might be related to the fact that we default to a release build here: https://github.com/jamulussoftware/jamulus/blob/aeb9375d97c2ade427143d7f39de02e4ab647b8d/Jamulus.pro#L26
Maybe using CONFIG-=release would help? Haven't tried it though

Maybe something in this documentation could also be helpful?
https://doc.qt.io/qtcreator/creator-build-settings.html

Other than that, I'm out of ideas and letting others chime in who do have the relevant experience.

**Reply by SamueleStefanoni** [reply]
Yes Christian!
Done it! Of course you meant to replace "release" with "debug". I understood it was a typo :)
Thanks thanks thanks

**Reply by hoffie** [reply]
> Of course you meant to replace "release" with "debug"

Replacing it with debug should work as well, yes (just make sure you don't accidentially commit the change ;)). I was hoping to be able to override it with `CONFIG-=release` (note the `-`), but I have never used this syntax and don't know if it works like that in general (or if it works like that from Qt creator).

Glad you've found a solution. Looking forward to your contributions! Any specific things you intend to work on? Might be good to submit potential ideas as a discussion first in order for things to settle instead of having to modify Pull Requests over and over again. :)

The CONTRIBUTING.md file is currently being reworked in #1379. The current state is at https://github.com/jamulussoftware/jamulus/blob/a82aaae00ec299c799bceb8755c3ca34e40265ea/CONTRIBUTING.md.

**Reply by SamueleStefanoni** [reply]
Ah Christian! Now I noted the `-` character, ok now I get it.
I would like to work on a topic that I already discussed in #826, but I would like to dive a little bit further, as the Lounge seems to me not really effective or not working. 
So my idea was to create a friend list so I can add my buddies, see if they are online or see in which server they are playing and of course to chat privately with them. 
Some issues may arise with name registration which is not actually an account so for this reason I was trying to debug and dive into the code. What do you think? Might be helpful?
